### PR TITLE
Fix #10: Full database export consume to much memory

### LINKED ISSUE #10: ErrorException: Allowed memory size: Full export of database
Calling an "export type 1" for long duration break on over memory consumption.

https://geokrety.org/export.php?modifiedsince=20000510032102&kocham_kaczynskiego=********

https://sentry.kumy.org/geokrety/geokrety-legacy/issues/547/

```
ErrorException: Allowed memory size of 134217728 bytes exhausted (tried to allocate 123015360 bytes)
  File "db.php", line 191, in handleFatalError
    $result = $this->mysqli->query($sql);
  File "templates/sentry-php-master/lib/Raven/ErrorHandler.php", line 0
    Raven_ErrorHandler->handleFatalError
```

### PR DESCRIPTION
**Pull request by kumy**
Let's use this PR, so everyone can familiarize with github review process.

### CODE REVIEW

**Comment by boly38** [comment]
direct output seems better than what we have.

to be security safe I would suggest to limit the sql query to X (x=1000) + accept pagination input (offset+limit) to avoid having geokrety server parsing 60k Geokrets by error but maybe 60k is not so much in term of performance. I don't know.

anyway 🎉 gogogo 😁

**Comment by kumy** [comment]
Keep in mind that the purpose is generate download dumps ;) exposed here https://cdn.geokrety.org/exports/ and referenced on the [API](https://geokrety.org/api.php) page.

**Comment by boly38** [comment]
what does travis ci do ? static analysis of the code ? (I m on mobile phone) will check that latrr anyway

**Comment by kumy** [comment]
@boly38 also please define your "review status" using the green "review changes" button on the "Files changed" tab :)

**Comment by kumy** [comment]
`Travis` check:
* phplint in php versions 5.6, 7.1, 7.2
* line ending by LF and no CRLF
* end of line white spaces.

The static analysis is done by [codacy](https://app.codacy.com/app/GeoKrety/geokrety-website/dashboard) - which BTW has not detected this PR :-1: 

@boly38 I've added you as an admin on codacy, will also do it for crowdin.

**Comment by kumy** [comment]
> Keep in mind that the purpose is generate download dumps ;) exposed here https://cdn.geokrety.org/exports/ and referenced on the API page.


> avoid having geokrety server parsing 60k Geokrets by error

I just remember that `export*.php` are also usable by anyone. But without the `token`, output are limited to 10 days of updated entries. Which is for the current audience quite small :smile:

File: export.php [hunk 1]
```diff
@@ -70,6 +70,15 @@ function ikonw($zmienna, $encoding)
 
 $now = date('Y-m-d H:i:s');
 
+header('Content-Type: application/xml');
+echo "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\" ?>
+<gkxml version=\"1.0\" date=\"$now\">";
```
Comment (romain-dartigues) [inline export.php]: I dislike abusive escaping, and I suppose [`short-open-tag`](http://php.net/short-open-tag) is ON, so why not:
```php
printf(
  '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<gkxml version="1.0" date="%s">',
  $now
);
```


I'll make you regret to ask for my input™.

Comment (kumy) [reply export.php]: > "I'll make you regret to ask for my input™"

:angry:

**Pull request merged**